engine/src/flutter/lib/ui/painting/rsuperellipse.cc

```diff
@@ -6,54 +6,67 @@
 
 #include "flutter/fml/logging.h"
 #include "third_party/tonic/logging/dart_error.h"
-#include "third_party/tonic/typed_data/typed_list.h"
 
-using flutter::RSuperellipse;
+namespace flutter {
 
-namespace tonic {
+using impeller::Scalar;
 
-// Construct an DlRoundSuperellipse from a Dart RSuperellipse object.
-// The Dart RSuperellipse is a Float32List containing
-//   [left, top, right, bottom, xRadius, yRadius]
-RSuperellipse DartConverter<flutter::RSuperellipse>::FromDart(
-    Dart_Handle value) {
-  Float32List buffer(value);
+IMPLEMENT_WRAPPERTYPEINFO(ui, RSuperellipse);
 
-  RSuperellipse result;
-  result.is_null = true;
-  if (buffer.data() == nullptr) {
-    return result;
+RSuperellipse::RSuperellipse(const tonic::Float64List& values) {
+  for (size_t i = 0; i < kValueCount; i++) {
+    values_[i] = values[i];
   }
+}
+
+RSuperellipse::~RSuperellipse() = default;
+
+flutter::DlRoundSuperellipse RSuperellipse::rsuperellipse() const {
+  return flutter::DlRoundSuperellipse::MakeRectRadii(bounds(), radii());
+}
 
+double RSuperellipse::getValue(int index) const {
+  if (index < 0 || index >= kValueCount) {
+    return 0;
+  }
+  return values_[index];
+}
+
+bool RSuperellipse::contains(double x, double y) const {
+  return param().Contains(
+      DlPoint(static_cast<Scalar>(x), static_cast<Scalar>(y)));
+}
+
+impeller::Scalar RSuperellipse::value32(int index) const {
+  return static_cast<Scalar>(getValue(index));
+}
+
+flutter::DlRect RSuperellipse::bounds() const {
   // The Flutter rect may be inverted (upside down, backward, or both)
   // Historically, Skia would normalize such rects but we will do that
   // manually below when we construct the Impeller RoundRect
   flutter::DlRect raw_rect =
-      flutter::DlRect::MakeLTRB(buffer[0], buffer[1], buffer[2], buffer[3]);
+      flutter::DlRect::MakeLTRB(value32(0), value32(1), value32(2), value32(3));
```

Comment:
These magic indexing numbers should have consts (or an enum) associated with them.